coral/src/app/features/topics/acl-request/forms/TopicProducerForm.tsx

```diff
@@ -163,15 +175,39 @@ const TopicProducerForm = ({
 
         <Grid cols={"2"} colGap={"4"} width={"fit"}>
           <GridItem>
-            <SubmitButton loading={isLoading}>Submit</SubmitButton>
+            <SubmitButton loading={isLoading}>Submit request</SubmitButton>
           </GridItem>
           <GridItem>
-            <SecondaryButton disabled={isLoading} onClick={() => navigate(-1)}>
+            <SecondaryButton
+              disabled={isLoading}
+              type="button"
+              onClick={
+                topicProducerForm.formState.isDirty
+                  ? () => setCancelDialogVisible(true)
+                  : () => cancelRequest()
+              }
+            >
               Cancel
             </SecondaryButton>
           </GridItem>
         </Grid>
       </Form>
+      {cancelDialogVisible && (
+        <Dialog
+          title={"Cancel ACL request?"}
+          primaryAction={{
+            text: "Cancel request",
+            onClick: () => cancelRequest(),
+          }}
+          secondaryAction={{
+            text: "Continue with request",
+            onClick: () => setCancelDialogVisible(false),
+          }}
+          type={"warning"}
+        >
+          <>Do you want to cancel this request? The data added will be lost.</>
```

Comment:
The fragments are used because `Dialog` doesn't take plain strings as children? :o

Reply:
Yep! I used the same typing as `Modal` here. Should we rather use `string` as a type for children here? 
Makes sense, hu? DS does that too as far as I can see.

Reply:
Updated! :D

Reply:
No, I thought that the type would be `React.ReactNode` and then you can either pass a string or DOM elements, but I don't think it's a big deal ^^Just curious!